backends/ebpf/codeGen.cpp

```diff
@@ -607,8 +612,73 @@ void CodeGenInspector::emitTCAssignmentEndianessConversion(const IR::Expression
         //
         emitAndConvertByteOrder(rexpr, "NETWORK"_cs);
     }
+    if (needsBitAlignment) {
+        builder->append("))");
+    }
+}
 
-    return;
+bool CodeGenInspector::storeBitAlignment(const IR::Type *ltype, const IR::Expression *lexpr,
+                                         cstring lpath) {
+    auto tcTarget = dynamic_cast<const P4TCTarget *>(builder->target);
+    if (lexpr != nullptr) {
+        if (!(lexpr->is<IR::Member>() || lexpr->is<IR::PathExpression>())) {
+            return false;
+        }
+    }
+    auto ebpfType = EBPFTypeFactory::instance->create(ltype);
+    EBPFScalarType *scalar = nullptr;
+    if (ebpfType->is<EBPFScalarType>()) {
+        scalar = ebpfType->to<EBPFScalarType>();
+        bool primitive = tcTarget->isPrimitiveByteAligned(scalar->implementationWidthInBits());
+        if (primitive) {
+            return false;
+        } else {
+            cstring storePrimitive = scalar->implementationWidthInBits() < 32
+                                         ? "storePrimitive32"_cs
+                                         : "storePrimitive64"_cs;
+            builder->appendFormat("%v((u8 *)&", storePrimitive);
+            if (lexpr != nullptr) {
+                visit(lexpr);
+            } else {
+                builder->append(lpath);
+            }
+            builder->appendFormat(", %d, (", scalar->implementationWidthInBits());
+            return true;
+        }
+    }
+    return false;
+}
+
+void CodeGenInspector::getBitAlignment(const IR::Expression *expression) {
+    if (expression->is<IR::Member>() || expression->is<IR::PathExpression>()) {
+        auto ftype = typeMap->getType(expression);
+        if (!ftype) {
+            visit(expression);
+            return;
+        }
+        auto tcTarget = dynamic_cast<const P4TCTarget *>(builder->target);
+        auto ebpfType = EBPFTypeFactory::instance->create(ftype);
+        EBPFScalarType *scalar = nullptr;
+        if (ebpfType->is<EBPFScalarType>()) {
+            scalar = ebpfType->to<EBPFScalarType>();
+            bool isPrimitive =
+                tcTarget->isPrimitiveByteAligned(scalar->implementationWidthInBits());
+            if (!isPrimitive) {
+                cstring getPrimitive = scalar->implementationWidthInBits() < 32
+                                           ? "getPrimitive32"_cs
+                                           : "getPrimitive64"_cs;
+                builder->appendFormat("%v((u8 *)", getPrimitive);
+                visit(expression);
+                builder->appendFormat(", %d)", scalar->implementationWidthInBits());
```

Comment:
```suggestion
                builder->appendFormat(", %d)", scalar->implementationWidthInBits());
                return;
```
By adding return to the above code, you can replace all else blocks below with just one visit(expression).